.github/workflows/check.yaml

```diff
@@ -0,0 +1,93 @@
+name: Check
+
+on:
+  pull_request:
+    branches: [ main ]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  base:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout change
+        uses: actions/checkout@v2
+        with:
+          path: permian
+
+      - name: Build main container
+        run: |
+          pushd $GITHUB_WORKSPACE/permian
+          ./build_container
+          popd
+
+      - name: Run base tests
+        run: |
+          pushd $GITHUB_WORKSPACE/permian
+          ./in_container make test
+          popd
+
+      - name: Build documentation
+        run: |
+          pushd $GITHUB_WORKSPACE/permian
+          ./in_container make doc
+          popd
+
+      - name: Cleanup
+        if: always()
+        run: |
+          podman rmi -f permian
+
+  secrets-check:
+    name: Check if some secrets are set
+    runs-on: ubuntu-latest
+    outputs:
+      extra_plugins_available: ${{ steps.extra_plugins.outputs.extra_plugins_available }}
+    steps:
+      - name: Check if ADDITIONAL_PLUGINS_REPO is set
+        id: extra_plugins
+        run: |
+            echo "::set-output name=extra_plugins_available::${{ secrets.ADDITIONAL_PLUGINS_REPO != '' }}"
+
+  extra-plugins:
+    needs: [secrets-check]
+    runs-on: [self-hosted, kstest]
+    if: needs.secrets-check.outputs.extra_plugins_available == 'true'
```

Comment:
Is there a reason why not to do:
```
 if: secrets.ADDITIONAL_PLUGINS_REPO != ''
```

Reply:
I tried it and it doesn't know secrets at that point. But I did it slightly differently so I can try again.

Reply:
Nope, doesn't work https://github.com/velezd/permian/actions/runs/1783600171

Reply:
I worried it is this reason. I'll try to dig a bit about that...